bee-api/bee-rest-api/src/endpoints/routes/api/plugins/debug/white_flag.rs

```diff
@@ -1,115 +1,69 @@
-pub(crate) fn filter<B: StorageBackend>(
-    public_routes: Box<[String]>,
-    allowed_ips: Box<[IpAddr]>,
-    storage: ResourceHandle<B>,
-    tangle: ResourceHandle<Tangle<B>>,
-    bus: ResourceHandle<Bus<'static>>,
-    message_requester: MessageRequesterWorker,
-    requested_messages: ResourceHandle<RequestedMessages>,
-    rest_api_config: RestApiConfig,
-) -> BoxedFilter<(impl Reply,)> {
-    self::path()
-        .and(warp::post())
-        .and(has_permission(ROUTE_WHITE_FLAG, public_routes, allowed_ips))
-        .and(warp::body::json())
-        .and(with_storage(storage))
-        .and(with_tangle(tangle))
-        .and(with_bus(bus))
-        .and(with_message_requester(message_requester))
-        .and(with_requested_messages(requested_messages))
-        .and(with_rest_api_config(rest_api_config))
-        .and_then(white_flag)
-        .boxed()
+pub(crate) fn filter<B: StorageBackend>() -> Router {
+    Router::new().route("/whiteflag", post(white_flag::<B>))
 }
 
 pub(crate) async fn white_flag<B: StorageBackend>(
-    body: JsonValue,
-    storage: ResourceHandle<B>,
-    tangle: ResourceHandle<Tangle<B>>,
-    bus: ResourceHandle<Bus<'static>>,
-    message_requester: MessageRequesterWorker,
-    requested_messages: ResourceHandle<RequestedMessages>,
-    rest_api_config: RestApiConfig,
-) -> Result<impl Reply, Rejection> {
+    Json(body): Json<Value>,
+    Extension(args): Extension<ApiArgsFullNode<B>>,
+) -> Result<impl IntoResponse, ApiError> {
     let index_json = &body["index"];
     let parents_json = &body["parentMessageIds"];
 
     let index = if index_json.is_null() {
-        return Err(reject::custom(CustomRejection::BadRequest(
+        return Err(ApiError::BadRequest(
             "Invalid index: expected a MilestoneIndex".to_string(),
-        )));
+        ));
     } else {
-        MilestoneIndex(index_json.as_u64().ok_or_else(|| {
-            reject::custom(CustomRejection::BadRequest(
-                "Invalid index: expected a MilestoneIndex".to_string(),
-            ))
-        })? as u32)
+        MilestoneIndex(
+            index_json
+                .as_u64()
+                .ok_or_else(|| ApiError::BadRequest("Invalid index: expected a MilestoneIndex".to_string()))?
+                as u32,
+        )
     };
 
     let parents = if parents_json.is_null() {
-        return Err(reject::custom(CustomRejection::BadRequest(
+        return Err(ApiError::BadRequest(
             "Invalid parents: expected an array of MessageId".to_string(),
-        )));
+        ));
     } else {
-        let array = parents_json.as_array().ok_or_else(|| {
-            reject::custom(CustomRejection::BadRequest(
-                "Invalid parents: expected an array of MessageId".to_string(),
-            ))
-        })?;
+        let array = parents_json
+            .as_array()
+            .ok_or_else(|| ApiError::BadRequest("Invalid parents: expected an array of MessageId".to_string()))?;
```

Comment:
There's a lot of repetition of this `BadRequest` message. You might want to store it in a constant. I don't know if the Rust compiler is smart enough to deduplicate those strings. @pvdrz ?

Reply:
I think it isn't, static string slices are tricky tho. Maybe `ApiError` should have an `invalid_parents()` constructor method

Reply:
Not sure if storing in a constant is really so nice or if it's better to live with the two duplicate strings.
I actually like the idea from @pvdrz however for consistency we would need to do this for all other errors too.
I decided against it when implementing it because we would end up with a quite a big error enum with lots of variants probably that couldn't be really reused as these errors are quite special